kafka-producer-java/src/main/java/Server.java

```diff
@@ -125,7 +123,13 @@ public void handle(HttpExchange exchange) throws IOException {
                 RecordHeaders headers = new RecordHeaders();
                 while (keys.hasNext()) {
                     String key = keys.next();
-                    headers.add(key, headersJson.getString(key).getBytes());
+                    if (headersJson.has(key)) {
+                        try {
+                            headers.add(key, headersJson.getString(key).getBytes());
+                        } catch (JSONException e) {
```

Comment:
Why not just do the same null check? JSONException has a bigger scope  than that

Reply:
This is true, but there is not other way of testing if headerJson has a key that is null, when you `getString` it will throw NPE if the value is null, if you check it with `has` before, you can ensure that it is indeed null value and not another exception.

Reply:
you can do `headerJson.get(key)`

Reply:
```java
public Object get(String key) throws JSONException {
        if (key == null) {
            throw new JSONException("Null key.");
        }
        Object object = this.opt(key);
        if (object == null) {
            throw new JSONException("JSONObject[" + quote(key) + "] not found.");
        }
        return object;
    }
```

This is the implementation of the `get` method that `getString` uses internally, if the key is null it will throw JsonException